### PR TITLE
Simplify and combine docker images

### PR DESCRIPTION
**Pull request by robinkrens**
- Dockerfile based on osfr/ros image, instead of nvidia cuda (no need to apt-get all ros packages, saves quite some time)
- Bash script to create noetic and melodic docker images
- Merge of various docker files (I don't think special windows images are needed)

Hardware passthrough X11 acceleration untested due to lack of nvidia gpu on current setup @Deakensbubbles can you check this?

### CODE REVIEW

**Comment by Deakensbubbles** [comment]
Your docker files do not pull the current workspaces from github. Should this not also be added in the dockerfiles? Also, I think the bash files for running the docker environments are useful because there are lots of arguments you need to pass to enable hardware acceleration and usb pass through.

**Comment by robinkrens** [comment]
For developers it is normally better to have the repository outside of the container, binding the directory with -v: 
```
-v Create a bind mount. If you specify, -v /HOST-DIR:/CONTAINER-DIR, Docker
          bind mounts /HOST-DIR in the host to /CONTAINER-DIR in the Docker
          container.
```
That way, you can use host dir tooling.

**Comment by robinkrens** [comment]
Agreed on the run bash script that it is useful, but since you probably need to edit it, I just put it in the readme.

**Comment by Deakensbubbles** [comment]
I noticed that when trying to run standard ros commands the setup.bash file is inaccessible. you're able to make use of ros commands when you use a clean environment and therefore you need to first 'catkin clean' your workspace and 'catkin_make' or 'catkin build' the workspace from within the container. You could add this disclaimer in the README.

File: Dockerfiles/run-container.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+# Example script to run a container
+# Please adjust accordingly
+SENSEGLOVE_DEV0=/dev/tty0
```
Comment (Deakensbubbles) [inline Dockerfiles/run-container.sh:5]: Should this not be ttyACM0 ?

**Review by Deakensbubbles — COMMENTED** [review]
Looks good except for this one small thing. If this is intentional or otherwise intended consider this PR approved.

File: Dockerfiles/Melodic/Dockerfile [hunk 2]
```diff
@@ -1,18 +1,57 @@
...
+
+# nvidia-container-runtime
+ENV NVIDIA_VISIBLE_DEVICES all
+ENV NVIDIA_DRIVER_CAPABILITIES compute,utility
+
+# Install additional packages, including ros_control
+RUN apt-get update && apt-get install -y git \
+ && apt-get install -y ros-melodic-ros-control ros-melodic-ros-controllers \
+ && apt-get install -y python-catkin-tools \
+ && apt-get install -y build-essential
```
[the remark this answers is not in the document]
Comment (robinkrens) [reply Dockerfiles/Melodic/Dockerfile:57]: Intended: workflow is clone repo outside container, run container with binding, catkin_make & source

File: Dockerfiles/Noetic/Dockerfile [hunk 3]
```diff
@@ -1,19 +1,57 @@
...
+
+# nvidia-container-runtime
+ENV NVIDIA_VISIBLE_DEVICES all
+ENV NVIDIA_DRIVER_CAPABILITIES compute,utility
+
+# Install additional packages, including ros_control
+RUN apt-get update && apt-get install -y git \
+ && apt-get install -y ros-noetic-ros-control ros-noetic-ros-controllers \
+ && apt-get install -y python3-catkin-tools \
+ && apt-get install -y build-essential
```
[the remark this answers is not in the document]
Comment (robinkrens) [reply Dockerfiles/Noetic/Dockerfile:57]: Intended: workflow is clone repo outside container, run container with binding, catkin_make & source

**Review by Deakensbubbles — APPROVED** [review]

**Pull request merged**